### ISSUE TITLE
Clarification on paper/impl mismatch (encoder)?

### ISSUE THREAD

**Issue by Jogima-cyber** [danijar/dreamerv3#68]
Hi, sorry to disturb you again, I've been working hard on analyzing your implementation and building a mini-dreamerV3 to get a good hold of how dreamerV3 works. I just got a question regarding the input to the encoder. In the paper, the following is stated:
<img width="730" alt="Capture d’écran 2023-06-16 à 17 25 27" src="https://github.com/danijar/dreamerv3/assets/58483181/439ed979-f8c7-442d-88c3-c9991ed01365">
So the encoder should take as input the recurrent hidden state (and the current observation), called "deter" in the implementation.
But in the implementation, if I'm not mistaken, the encoder only takes as input the current observation. Do you confirm this or have I misunderstood something in the implementation?

**Comment by danijar** [comment]
That's just a naming mismatch. The encoder in the code only takes in observations but then passes its features to the RSSM where it gets combined with the RNN state before computing the logits for z_t from it.

**Comment by Jogima-cyber** [comment]
Oh my bad, sorry, I should have figured it out on my own, got stupidly confused. Thanks for the answer.

**Comment by danijar** [comment]
No worries at all, thanks for reaching out.
